core/src/main/java/org/aerogear/mobile/core/metrics/MetricsService.java

```diff
@@ -74,25 +43,60 @@ public void sendAppAndDeviceMetrics(final Callback callback) {
      * @param type type of the enclosing metrics event
      * @param metrics Metrics to send
      */
-    public void publish(String type, Metrics... metrics) {
-        publish(type, metrics, null);
+    public Request<Boolean> publish(@NonNull String type, @NonNull final Metrics... metrics) {
+
+        return Requester.call(() -> MobileCore.getInstance().getHttpLayer().newRequest())
```

Comment:
You should be able to use it like this : 

```
return MobileCore.getInstance().getHttpLayer().newRequest()
                                        .post(url, createMetricsJSONObject(type, metrics).toString().getBytes())
                                        .requestMap(httpResponse -> 
                                            switch (httpResponse.getStatus()) {
                                                case HTTP_OK:
                                                    return Boolean.TRUE;
                                                default:
                                                    throw (new HttpException(
                                                                    httpResponse.getStatus()));
                                            }
                                        ).requestOn(new AppExecutors().networkThread()));
```

I'll need to pop this into android studio and confirm the exact syntac, but you don't need to wrap everything in a request when you can just chain them.

Reply:
For some reason `.requestMap(httpResponse` from `.post` does not work for me.

Reply:
Ahhh yeah looks like RequestMapFunction has silly return type. that I need to fix